core/sugartypes.ml

```diff
@@ -218,6 +183,7 @@ and iterpatt = [
 ]
 and sec = [`Minus | `FloatMinus | `Project of name | `Name of name]
 and declared_linearity = [ `Lin | `Unl ]
+and fn_dep = string * string
```

Comment:
This doesn't need to be part of the mutual recursion (though I see other things above that also don't need to be)

Reply:
I'm happy to change this if it is preferred to be out of the mutual recursion.

Reply:
I seem to remember that type definition are recursive by default in OCaml. So
```ocaml
type t = int
and s = float
```
is only "marginally worse" than 
```ocaml
type t = int
type s = float
```
If you don't want a recursive type, you have to explicitly say so
```ocaml
type nonrec t = int
type nonrec s = float
```